Add markdown editor on conversations and comments

cmrd-senya opened the pull request:
This is a follow up for #7214.

I fixed all the notes by @svbergerem and rebased the code.

I didn't do thorough review or test myself, so this needs review. Maybe I'll do one myself if nobody else does it earlier.

christophehenry committed:
Add markdown editor on conversations and comments

christophehenry committed:
Fix CSS and blur on commant submission

Commit:
Fix autosize for the textarea in conversations

We need to reinit autosize when we navigate in conversations with the backbone router.

goobertron commented:
Great, thanks!

Commit:
MD editor coversations&comments: refactorings

Javascript refactorings for better maintainability:
* simplePreview function to avoid code duplication
* Preloaded conversation is processed in a slightly different way
* Split isPreviewOrTexareaNotEmpty function since it has more than one responsibility

cmrd-senya commented:
Okay, I reviewed the code and made some changes I found desirable. For me it looks good now. I think we need a one more review now, possibly from @svbergerem to merge it.

svbergerem commented on spec/javascripts/helpers/markdown_editor_spec.js:
```diff
@@ -220,6 +220,38 @@ describe("Diaspora.MarkdownEditor", function() {
+      this.target.instance = undefined;
+      expect(this.target.isPreviewMode()).toBe(false);
+    });
+
```
Please add `returns false if the editor is in write mode`.

svbergerem commented on spec/javascripts/helpers/markdown_editor_spec.js:
```diff
@@ -220,6 +220,38 @@ describe("Diaspora.MarkdownEditor", function() {
+      this.target.instance = undefined;
+      expect(this.target.userInputNotEmpty()).toBe(false);
+    });
+
```
As written above, I think this method should be called `userInputEmpty` and behave accordingly. Afterwards please add a test `returns true if the editor is empty`.

svbergerem reviewed (changes requested):
Just some minor remarks. Everything else looks good. Please squash the commits afterwards.

Commit:
fixup! MD editor coversations&comments: refactorings

cmrd-senya commented:
I squashed and rebased everything that was done before your notes. For your notes I added changes as a fixup commit so you could check what I changed exactly.

svbergerem commented:
I squashed the fixup commit while merging. Thank you! :)

Flaburgan commented:
Just pushed that change to diaspora-fr. Thank you two VERY MUCH for your work!

goobertron commented:
Yes, a HUGE thank you to @cmrd-senya and @svbergerem!

cmrd-senya commented:
And to @AugierLe42e who developed the original code!
